lang/c++/impl/Compiler.cc

```diff
@@ -263,7 +266,7 @@ static GenericDatum makeGenericDatum(NodePtr n,
 static const std::unordered_set<std::string>& getKnownFields() {
     // return known fields
     static const std::unordered_set<std::string> kKnownFields =
-        {"name", "type", "default", "doc", "size", "logicalType",
+        {"name", "type", "aliases", "default", "doc", "size", "logicalType",
```

Comment:
As well as being necessary for this change, adding `"aliases"` to this list fixes what is - I would argue - a regression in `master` vs `1.11` where any type or field aliases would now result in an error. This is because the custom attribute handling would not recognise it as a known field and would then try and read a string value. This fails because the `aliases` value is, of course, an array of strings. I don't know if there are any other features in the specification that are currently unsupported and would be affected similarly. I think that you should make this change even if you don't accept the rest of the PR.